### PR TITLE
[MM-53851] Add limiting ability to log fields

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We've added a field limiting ability to the logging library so that user provided values would not flood the logs. The default value is `2048` but it's purely a shot in the dark. The fields that exceed the limit will be truncated and a 3-dot will be added to indicate that field has been truncated.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53851

#### Screenshots
```bash
{"timestamp":"2023-08-11 21:46:44.745 +03:00","level":"info","msg":"Starting Server...","caller":"app/server.go:914"}
{"timestamp":"2023-08-11 21:46:44.745 +03:00","level":"info","msg":"Server is listening on [::]:8065","caller":"app/server.go:990","address":"[::]:8065"}
{"timestamp":"2023-08-11 21:46:47.033 +03:00","level":"warn","msg":"error fetching image","error":"Get \"http://localhost: this is a very lo..."}
```

#### Release Note

```release-note
A new config value added to LogSettings so that log fields can be size limited during logging.
```

### CODE REVIEW

File: server/channels/app/platform/log.go [hunk 1]
```diff
@@ -31,7 +32,7 @@ func (ps *PlatformService) initLogging() error {
 	// create the app logger if needed
 	if ps.logger == nil {
 		var err error
-		ps.logger, err = mlog.NewLogger()
+		ps.logger, err = mlog.NewLogger(logr.MaxFieldLen(*ps.Config().LogSettings.MaxFieldSize))
```
Comment (agnivade) [inline server/channels/app/platform/log.go:35]: Ideally, this should be a part of `mlog.LoggerConfiguration`, but I see that it's actually just a target configuration. I'd prefer if we change that to be a struct and add the `MaxFieldSize` there as well. But don't want to block the PR. Could be made as a future improvement.

**Review by agnivade — APPROVED** [review]

**Comment by isacikgoz** [comment]
> As a piece of meta feedback, would have been nice to review the full context

Thanks for the feedback @streamer45, here is the link for the original `logr` PR: https://github.com/mattermost/logr/pull/40/files and another link for a bugfix: https://github.com/mattermost/logr/pull/41/files

File: server/channels/app/platform/log.go (hunk 1 above)
Comment (isacikgoz) [reply server/channels/app/platform/log.go:35]: Added the ticket: https://mattermost.atlassian.net/browse/MM-54122

File: server/public/model/config.go [hunk 2]
```diff
@@ -1343,6 +1344,10 @@ func (s *LogSettings) SetDefaults() {
 	if s.AdvancedLoggingConfig == nil {
 		s.AdvancedLoggingConfig = NewString("")
 	}
+
+	if s.MaxFieldSize == nil {
+		s.MaxFieldSize = NewInt(2048)
```
[the remark this answers is not in the document]
Comment (jupenur) [reply server/public/model/config.go:1349]: I checked with SecOps; we don't have enough historical log data to do a meaningful analysis. The 2k default seems relatively safe to me, though, and if it turns out there are issues we can just adjust it later.

**Review by jupenur — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed LGTM!

- I have tested this to ensure it’s functional. The new field works and the user generated string as part of the log record is concatenated.
- Tested various values but upper boundaries larger than 1000000 characters are not really applicable as the size of the payload eventually makes the request time out and then the logging is not applicable. 
- For cases where the string is written to the server log the truncation is working as expected.

**Pull request merged**